client/masterAgent2.py

```diff
@@ -52,63 +59,138 @@ def assignGoals(self, agentsToReplan):
+                    agent_position = self.currentState.find_agent(agent.name)
+                    '''
+                        We check if the agent and the goal are connected, if not we search for another agent
+                        Right now we know:
+                            - which box can fill the goal
+                            - the agent can move thoses boxes
+
+                        As we already have checked if the boxes_able_to_fill_goal are connected to the goal,
+                        we will not need to check if agent and box are connected
+                    '''
+                    if not self.currentState.check_if_connected(agent_position, prioritized_goal['position']):
+                        # print('Goal:', goal['position'],'is not connected with the agent', 'agent', agent_position, file=sys.stderr, flush=True)
+                        continue
+                    
+                    ''' We can add store the information that agent can achieve this goal'''
+                    agents_connected.append((agent, agent_position))
+
+                    ''' For the box that can fill the goal, we'll try to assign a goal to the agent '''
+                    prioritizedBoxes = list(boxes_able_to_fill_goal)
+                    while prioritizedBoxes != [] and not prioritized_goal_is_assigned:
+                        (box, box_pos) = prioritizedBoxes.pop(0)
+
+                        # We avoid assigning a box that is already placed on a goal.
+                        boxAlreadyPlaced = False
+                        for goalmet in goalsMet:
+                            boxPlaced = self.currentState.find_box(goalmet['position'])
+                            if boxPlaced.variables[0] == box['name']:
+                                boxAlreadyPlaced = True
+
+                        if not boxAlreadyPlaced:
+                            if box_pos in agent.tracker.boundary:
+                                box_tracker = None
+                                # We initiate a tracker at the box position
+                                if str(box_pos) in box_tracker_dict:
+                                    box_tracker = box_tracker_dict[str(box_pos)]
+                                else:
+                                    box_tracker = Tracker(box_pos)
+                                    box_tracker_dict[str(box_pos)] = box_tracker
+                                    box_tracker.estimate(self.currentState)
+
+                                # Then we see if the goal is reachable from the box or if the goal is reachable from agent
+                                # If yes --> we can assign a goal
+                                box_can_reach_goal = prioritized_goal['position'] in box_tracker.reachable
+                                agent_can_reach_goal = prioritized_goal['position'] in agent.tracker.reachable
+                                if box_can_reach_goal or agent_can_reach_goal:
```

Comment:
Everything looks really good 👍🏽But I was just wondering why on this line we have "if box_can_reach_goal **or** agent_can_reach_goal"? shouldn't it be an **and** instead?

Reply:
No it's or, just imagine `0   A a`.  The agent can not reach the goal but the box can so it's ok